### DISCUSSION TITLE
Errors in Rust memory game example?

### DISCUSSION THREAD

**Discussion by troelsarvin** [slint-ui/slint#2493] [General]
In the code at https://slint-ui.com/releases/1.0.0/docs/tutorial/rust/polishing_the_tile.html I had to change 

```
    // Left curtain
    Rectangle {
        background: #193076;
        width: open_curtain ? 0px : (parent.width / 2);
        height: parent.height;
        animate width { duration: 250ms; easing: ease-in; }
    }
```

to 

```
    // Left curtain
    Rectangle {
        background: #193076;
        width: open_curtain ? 0px : parent.width; // CHANGED
        height: parent.height;
        animate width { duration: 250ms; easing: ease-in; }
    }
```

Otherwise, the tiles would look like this when the application starts:
![sshot_partically_closed_curtains](https://user-images.githubusercontent.com/5257280/230148905-b56074c4-7a8b-47b2-bdef-bb49a360a0a7.png)

This is on a Fedora Linux 37.

The code at https://slint-ui.com/releases/1.0.0/docs/tutorial/rust/creating_the_tiles_from_rust.html makes Rust refuse to compile:

```
error[E0624]: associated function `get_memory_tiles` is private
   --> src/main.rs:7:48
    |
7   |       let mut tiles: Vec<TileData> = main_window.get_memory_tiles().iter().collect();
    |                                                  ^^^^^^^^^^^^^^^^ private associated function
...
23  | / slint::slint! {
24  | |     struct TileData {
25  | |         image: image,
26  | |         image_visible: bool,
...   |
100 | |     }
101 | | }
    | |_- private associated function defined here

error[E0624]: associated function `set_memory_tiles` is private
   --> src/main.rs:18:17
    |
18  |       main_window.set_memory_tiles(tiles_model.into());
    |                   ^^^^^^^^^^^^^^^^ private associated function
...
23  | / slint::slint! {
24  | |     struct TileData {
25  | |         image: image,
26  | |         image_visible: bool,
...   |
100 | |     }
101 | | }
    | |_- private associated function defined here

warning: unused import: `slint::Model`
 --> src/main.rs:2:9
  |
2 |     use slint::Model;
  |         ^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

warning: unused import: `rand::seq::SliceRandom`
  --> src/main.rs:12:9
   |
12 |     use rand::seq::SliceRandom;
   | 
```

This is with Rust 1.68.2.

**Comment by donbright** [comment]
I'm on rustc 1.66.1 (90743e729 2023-01-10) and i cant get this thing to compile either

i got through the thing where i can see multiple items and open and close them

but when it comes to 'checking if there is more than one open and preventing that', its not compiling
 

What i found as a workaround --- just

```
    git clone https://github.com/slint-ui/slint.git
    cd examples/memory 
    cargo run
```

you can do the same in the other directories like energy-monitor. i even got a wrning that QT was the wrong version but it still ran anyways. i dont even know how it works. but it is definitely worth pushing through the issues because it looks really quite nice.

this also works:

```
    git clone https://github.com/slint-ui/slint.git
    cargo run --bin memory
```

what does not work is the standard `cargo run --example memory` but thats not too big a deal

note - it also takes like, a few gigs of RAM during build and like maybe 8-ish gigabytes of disk space for all dependencies. 

the tutorial appears broken, im thinking maybe the tutorial could be somehow integrated with the example directory so it would be part of the continuous integration/testing. not sure of the best way to do that tho

**Reply by tronical** [reply]
Hi @donbright . I just went through the tutorial once again and it's compiling fine for me here. The issue that @troelsarvin ran into was fixed (you can see that in the updated content the link refers to).

I'm curious, what compile issues are you running into? Could it be that the instructions in the last step about inserting the code are a little unclear? (would like to improve them)
